### DISCUSSION TITLE
Site to site windows <-> strongswan (tunnel-mode, ikev2 with pki and eap authentication)

### DISCUSSION THREAD

**Discussion by jfossheim-skyfritt** [strongswan/strongswan#2105] [General]
I am looking into setting up site-to-site connection with a windows 10 machine (hypervisor) and a strongswan-based gateway. No virtual IPs  (just one subnet to another). And I am looking for some insights and alternatives, on how to do the configuration on the windows side. 

I have successfully configured this from a linux-box with strogswan to the gateway. Is there away to do this with the built in infrastructure that are shipped with windows (10/11)?

I have tried to look into powershell conmandlets, but still no obvious paths to what I want to achieve. I have to use PKI and EAP-authentication.

I appreciate any tips or insights when it comes to configuring windows. 

I read with interest that strongswan that strongswan can be compiled "natively" on windows? Without beeing able to use virtual IPs, which dosen’t matter in this setup anyway! 

https://docs.strongswan.org/docs/5.9/os/windows.html

Is there any precompiled binary somewhere? Or is compiling myself the only option?

**Comment by Incises** [comment]
You may install [MSYS2](https://www.msys2.org/#installation) on Windows to compile the 'native' version. After the installation, install the gcc toolchains on it and follow the cross-compile [options](https://docs.strongswan.org/docs/5.9/os/windows.html#_toolchain) in the MSYS2-mingw64 shell.

**Comment by tobiasbrunner** [comment]
> Is there away to do this with the built in infrastructure that are shipped with windows (10/11)?

I don't think the built-in IKEv2 client supports net-to-net setups (I guess that requires a Windows Server). However, the old IKEv1 client that can be found in the advanced firewall might (see [this old wiki page](https://wiki.strongswan.org/projects/strongswan/wiki/WindowsSuiteB) for some pointers). But I generally wouldn't recommend using IKEv1.

> Is there any precompiled binary somewhere? Or is compiling myself the only option?

No, there are no binaries, you'll have to compile it yourself (either on Windows via MSYS2 or cross-compile on a Linux host, see [test.sh](https://github.com/strongswan/strongswan/blob/187c72d1afdcc4d214a3bcabb1c95c93ea1be21e/scripts/test.sh#L309-L314) for pointers on required packages).
